pkg/operator/configobservation/node/observe_latency_profile.go

```diff
@@ -0,0 +1,127 @@
+package node
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/configobservation"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	configV1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+)
+
+var notReadyTolerationSecondsPath = []string{"apiServerArguments", "default-not-ready-toleration-seconds"}
+var unreachableTolerationSecondsPath = []string{"apiServerArguments", "default-unreachable-toleration-seconds"}
+
+// ObserveNotReadyTolerationSeconds observes the value that should be set for default-not-ready-toleration-seconds
+// api server argument on the basis of provided worker latency profile from config node object.
+func ObserveNotReadyTolerationSeconds(genericListers configobserver.Listers, _ events.Recorder, existingConfig map[string]interface{}) (ret map[string]interface{}, errs []error) {
+	defer func() {
+		// Prune the observed config so that it only contains default-not-ready-toleration-seconds field.
+		ret = configobserver.Pruned(ret, notReadyTolerationSecondsPath)
+	}()
+
+	nodeLister := genericListers.(configobservation.Listers).NodeLister
+	node, err := nodeLister.Get("cluster")
+	if err != nil && !apierrors.IsNotFound(err) {
+		// we got an error so without the node object we are not able to determine worker latency profile
+		return existingConfig, append(errs, err)
+	}
+
+	// read the observed value
+	var observedDefaultNotReadyTolerationSeconds string
+	switch {
+	case node.Spec.WorkerLatencyProfile == configV1.DefaultUpdateDefaultReaction:
+		observedDefaultNotReadyTolerationSeconds = strconv.Itoa(configV1.DefaultNotReadyTolerationSeconds)
+	case node.Spec.WorkerLatencyProfile == configV1.MediumUpdateAverageReaction:
+		observedDefaultNotReadyTolerationSeconds = strconv.Itoa(configV1.MediumNotReadyTolerationSeconds)
+	case node.Spec.WorkerLatencyProfile == configV1.LowUpdateSlowReaction:
+		observedDefaultNotReadyTolerationSeconds = strconv.Itoa(configV1.LowNotReadyTolerationSeconds)
+	// in case of empty worker latency profile, do not update config
+	case node.Spec.WorkerLatencyProfile == "":
+		return existingConfig, errs
+	}
```

Comment:
what if we add a new profile in the future?

Reply:
Similar to: https://github.com/openshift/cluster-kube-apiserver-operator/pull/1328#discussion_r854181024.